dts/arm/st/stm32f1.dtsi

```diff
@@ -111,6 +111,26 @@
 			status = "disabled";
 			label = "SPI_1";
 		};
+
+		spi2: spi@40003800 {
+			compatible = "st,stm32-spi";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0x40003800 0x400>;
+			interrupts = <36 5>;
+			status = "disabled";
+			label = "SPI_2";
+		};
+
+		spi3: spi@40003C00 {
```

Comment:
SPI3 is not available on stm32f103xb. You  should move it inside stm32f103Xe.dtsi and stm32f107.dtsi.